int8 id
float32 angle
---
bool success